### ISSUE TITLE
Reporting for integration tests of multiproject builds is non-deterministic

### ISSUE THREAD

**Issue by Z1kkurat** [sbt/sbt-jacoco#90]
### Expected vs Actual Behaviour

In multiproject builds where multiple projects have integration tests, result of task `it:jacoco` is non-deterministic - you can observe any of subprojects randomly failing to meet requirements for some reason. Just like `fork in Test := true` with unit tests, it can be solved with `fork in IntegrationTest := true` but as it can be seen in [base plugin code](https://github.com/sbt/sbt-jacoco/blob/master/src/main/scala/com/github/sbt/jacoco/BaseJacocoPlugin.scala#L38), agent dependency is added only for `Test` configuration in case of forking, so running `it:jacoco` with `fork in IntegrationTest := true` leads to exceptions like this: `sbt.ForkMain$ForkError: java.lang.NoClassDefFoundError: org/jacoco/agent/rt/internal_8ff85ea/Offline`. I managed to overcome this by manually adding agent dependency in case we are forking for IT:
```scala
libraryDependencies ++= {
    if ((fork in IntegrationTest).value) {
      Seq("org.jacoco" % "org.jacoco.agent" % BuildInfo.jacocoVersion % IntegrationTest classifier "runtime")
    } else {
      Seq.empty[ModuleID]
    }
  }
```
If you don't mind, I can submit a PR with fix for it in `BaseJacocoPlugin`, changing  
`Seq("org.jacoco" % "org.jacoco.agent" % BuildInfo.jacocoVersion % Test classifier "runtime")`  
to  
`Seq("org.jacoco" % "org.jacoco.agent" % BuildInfo.jacocoVersion % srcConfig classifier "runtime")` )

Example of such manupulations can be seen at [my repo](https://github.com/Z1kkurat/sbt-jacoco-example/blob/sbt-jacoco-new/build.sbt). You can try to remove this `libraryDependency` or set `fork in IntegrationTest := false` and see errors.

### Environment

* SBT version: 1.0.0
* Plugin version: 3.0.1
* Scala version(s): 2.12.3
* Java version: 1.8.0_131
* Link to project source: https://github.com/Z1kkurat/sbt-jacoco-example/blob/sbt-jacoco-new/build.sbt

**Comment by stringbean** [comment]
I think I might know why this is - the JaCoCo data gets stored in the plugin during the tests. As the tests are getting run in parallel then weird things can happen.

I'l take a look this week and find a better solution.

**Comment by Z1kkurat** [comment]
@stringbean So, besides random test failing, there's still problem with `fork in IntegrationTest := true` which causes failure due to absence of jacoco agent. Do I need to create a separate issue for that, or just submit a PR, as I described in first post? I do think that plugin users shouldn't have trouble with forking during testing.

**Comment by stringbean** [comment]
@Z1kkurat: please raise a separate issue.

**Comment by stringbean** [comment]
Fixed in 3.0.2
